src/coreclr/tools/Common/Pgo/PgoFormat.cs

```diff
@@ -48,6 +48,12 @@ public enum PgoInstrumentationKind
         EdgeLongCount = (DescriptorMin * 6) | EightByte, // edge counter using unsigned 8 byte int
         GetLikelyClass = (DescriptorMin * 7) | TypeHandle, // Compressed get likely class data
         GetLikelyMethod = (DescriptorMin * 7) | MethodHandle, // Compressed get likely method data
+
+        // Same as type/method histograms, but for generic integer values
+        ValueHistogramIntCount = (DescriptorMin * 8) | FourByte | AlignPointer,
+        ValueHistogramLongCount = (DescriptorMin * 8) | EightByte,
+        ValueHistogram = (DescriptorMin * 9) | EightByte,
+        GetLikelyValue = (DescriptorMin * 10) | EightByte,
```

Comment:
These need proper handling in dotnet-pgo and crossgen2. Also, `GetLikelyValue` should either be removed or should be computed by crossgen2.

cc @davidwrighton, this is once again going to break backwards compatibility of a .mibc produced by a new dotnet-pgo and consumed by an old crossgen2. Do we need to do anything special here? Also, do we need to bump R2R minor version?

Reply:
@jakobbotsch I've removed the compressed `GetLikelyValue` for now, will implement it separately.

What kind of additional handling for dotnet-pgo and crossgen2 we talk about besides the compressed one?

I've validated that dotnet-pgo works properly, e.g.:

![image](https://github.com/dotnet/runtime/assets/523221/f359075e-a3bf-40bc-9d50-2d23b7574ae0)

(dotnet-pgo dump mibc)

Reply:
I think if you plan to implement it separately, it makes sense to add it now (and implementing the minimum format-handling stuff for it) so we don't need to update the format twice.

> What kind of additional handling for dotnet-pgo and crossgen2 we talk about besides the compressed one?

Seems like `PgoFormat.Merge` needs to be updated at least. Looks like the encoder/decoder (`EncodePgoData`/`ParsePgoData`) might be ok since this reuses `EightByte`. That's the one where backwards compatibility was broken when I added `MethodHandle` last time, so maybe this change doesn't break backwards compatibility.

Reply:
Yeah, I put my request changes in to cover this sort of thing. I think you might be OK, but what you need to do is validate that the new instrumentation doesn't cause problems with old dotnet-pgo/crossgen2. If it does, we can have another discussion, but the first thing to see is if it doesn't. The MIBC format is designed to allow new things to appear, but this only works in a subset of ways that it is possible to add new content to the file, and @jakobbotsch didn't follow those restrictions with the last PGO data format change he made. (Honestly, though, that was my fault, as I'd forgotten about them.)

Reply:
Oh, and when you do this validation, please write down what scenarios you tested. I'll probably want to add a few weird wrinkles to it all.

Reply:
I updated dotnet-pgo (merge and dump) and SPMI to be aware of value probes but it looked like they were fine as is too (didn't crash).

also, I've done some testing that @davidwrighton requested and published my results and steps via mail